File: Output/Utility/Hooks/use-orientation.hook.lua
-- Compiled with roblox-ts v2.2.0
local TS = _G[script]
local useViewport = TS.import(script, TS.getModule(script, "@rbxts", "pretty-roact-hooks").out).useViewport
local useState = TS.import(script, TS.getModule(script, "@rbxts", "roact-hooked").src).useState
local function useOrientation()
	local Orientation, SetOrientation = useState("Landscape")
	useViewport(function(Viewport)
		SetOrientation(if (Viewport.Y > Viewport.X) then "Portrait" else "Landscape")
	end)
	return Orientation
end
return {
	useOrientation = useOrientation,
}

File: Source/Utility/Hooks/use-orientation.hook.ts
import { useViewport } from "@rbxts/pretty-roact-hooks";
import { useState } from "@rbxts/roact-hooked";

export function useOrientation() {
	const [ Orientation, SetOrientation ] = useState<"Landscape" | "Portrait">("Landscape");

	useViewport(Viewport => {
		SetOrientation((Viewport.Y > Viewport.X) ? "Portrait" : "Landscape");
	});

	return Orientation;
};